packages/puppeteer-core/src/common/WaitTask.ts

```diff
@@ -170,11 +170,10 @@ export class WaitTask<T = unknown> {
   async terminate(error?: unknown): Promise<void> {
     this.#world.taskManager.delete(this);
 
-    if (this.#timeout) {
-      clearTimeout(this.#timeout);
-    }
-
     if (error && !this.#result.finished()) {
+      if (this.#timeout) {
```

Comment:
is this change needed?

Reply:
This could be a separate change, but I've made it here since it's small. Forgot to remark about it.